src/main/webapp/groups.js

```diff
@@ -0,0 +1,121 @@
+function redirectToResultsGroupsHelper(document, window) {
+    var group = document.getElementById("search-box-results").value;
+    
+    var url = "groups.html?group=" + encodeURIComponent(group);
+    window.location.href = url;
+
+    return false;
+}
+
+/** Gets the name the group searched for from the group search box and displays a list of the results from querying the databse
+    for the given group name. */
+function getSearchGroupResults() {
+    return getSearchGroupResultsHelper(document, window);
+}
+
+/** Helper function for getSearchGroupResults, used for testing purposes. */
+async function getSearchGroupResultsHelper(document, window) {
+    var queryString = new Array();
+    window.onload = readComponents(queryString, window);
+    var group = queryString["group"];
+
+    if(group != null) {
+        var groups = getGroups(group);
+        
+        await groups;
+    }
+}
+
+/** Fetches the list of groups for the group name the user searched for. */
+async function getGroups(group) {
+    await fetch("/manage-groups?group=" + group).then(response => response.json()).then((results) => {
+        var groupContainer = document.getElementById("groups");
+
+        var numSearchResults = document.createElement("h4");
+        numSearchResults.className = "num-search-results";
+
+        groupContainer.appendChild(numSearchResults);
+        
+        //if there was an error reported by the servlet, display the error message
+        if(results.error) {
+            numSearchResults.innerText = results.error;
+            return;
+        }
+
+        //Only make "groups" plural if there are 0 or more than 1 tutors
+        numSearchResults.innerText = "Found " + results.length + (results.length > 1 || results.length === 0 ? " groups named " : " group named ") + group;
+
+        results.forEach(function(result) {
+            groupContainer.append(createGroupResult(result));
+        });
+    });
+
+}
+
+/** Creates a div element containing information about a group result. */
+function createGroupResult(result) {
+    var container = document.createElement("div");
+    var name = document.createElement("h3");
+    name.style.textTransform = 'capitalize';
+    //var members = document.createElement("h4");
+    var topic = document.createElement("h4");
+    topic.style.textTransform = 'capitalize';
+    var description = document.createElement("h5");
+    var groupLink = document.createElement("a");
+
+    name.innerText = result.name;
+    //members.innerText = result.members + (result.members > 1 || result.members === 0 ? " members" : " member");
+    topic.innerText = result.topic;
+    description.innerText = result.description;
+    groupLink.innerText = "View";
+    groupLink.href = "/group.html?groupId=" + result.id;
+
+    container.classList.add("user-result");
+    container.classList.add("list-group-item");
+
+    container.appendChild(name);
+    //container.appendChild(members);
```

Comment:
Delete

Reply:
I left those there because I wanted to implement a members count in the future but I will just delete them for now